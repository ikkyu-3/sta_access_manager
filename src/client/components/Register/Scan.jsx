// @flow
import type { RegisterScanProps } from "@types";

import * as React from "react";
import SingleButton from "@components/SingleButton";
import styles from "./style.css";

class RegisterScan extends React.Component<RegisterScanProps> {
  componentDidMount() {
    this.props.onChangeStep();
  }

  componentWillUnmount() {
    this.props.unmount();
  }

  render() {
    const { onCancelButton } = this.props;
    const { main, textOnly, contentWithButton } = styles;
    return (
      <div className={main}>
        <div className={`${contentWithButton} ${textOnly}`}>
          カードをスキャンして下さい
        </div>
        <SingleButton
          text="戻る"
          className="default"
          onButtonClick={onCancelButton}
        />
      </div>
    );
  }
}
export default RegisterScan;
