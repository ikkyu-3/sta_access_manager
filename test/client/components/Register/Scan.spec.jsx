import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Scan from "@components/Register/Scan";

const fn = () => {};

describe("Register/Scan.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(
        <Scan onChangeStep={fn} unmount={fn} onCancelButton={fn} />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const onChangeStepMock = jest.fn();
    const unmountMock = jest.fn();

    let scan;

    beforeAll(() => {
      scan = shallow(
        <Scan
          onChangeStep={onChangeStepMock}
          unmount={unmountMock}
          onCancelButton={fn}
        />
      );
    });

    it("componentDidMountは、onChangeStepを実行する", () => {
      expect(onChangeStepMock).toBeCalled();
    });

    it("componentWillUnmountは、unmountを実行する", () => {
      scan.instance().componentWillUnmount();
      expect(unmountMock).toBeCalled();
    });
  });
});
