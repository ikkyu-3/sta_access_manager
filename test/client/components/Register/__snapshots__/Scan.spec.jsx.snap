// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register/Scan.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="main"
>
  <div
    className="contentWithButton textOnly"
  >
    カードをスキャンして下さい
  </div>
  <div
    className="singleButton"
  >
    <button
      className="default"
      onClick={[Function]}
    >
      戻る
    </button>
  </div>
</div>
`;
